core-bundle/src/Resources/contao/library/Contao/Search.php

```diff
@@ -782,33 +783,31 @@ public static function query(string $strKeywords, bool $blnOrSearch=false, array
 	 *
 	 * @param string $strUrl The URL to be removed
 	 */
-	public static function removeEntry($strUrl)
+	public static function removeEntry($strUrl, Connection $connection = null)
 	{
-		$objDatabase = Database::getInstance();
+		$connection = $connection ?? System::getContainer()->get('database_connection');
 
-		$objResult = $objDatabase->prepare("SELECT id FROM tl_search WHERE url=?")
-								 ->execute($strUrl);
+		$result = $connection
+			->prepare('SELECT id FROM tl_search WHERE url=:url')
+			->executeQuery(array('url' => $strUrl));
 
-		while ($objResult->next())
+		foreach ($result->fetchAllAssociativeIndexed() as $id)
```

Comment:
This doesn’t seem to work

```suggestion
		foreach ($result->fetchFirstColumn() as $id)
```